ProcessLib/IntegrationPointSerialization.h

```diff
@@ -0,0 +1,22 @@
+/**
+ * \file
+ *
+ * \copyright
+ * Copyright (c) 2012-2017, OpenGeoSys Community (http://www.opengeosys.org)
+ *            Distributed under a Modified BSD License.
+ *              See accompanying file LICENSE.txt or
+ *              http://www.opengeosys.org/project/license
+ *
+ */
+
+#pragma once
+
+namespace ProcessLib
+{
+struct IntegrationPointSerialization
+{
+    virtual std::size_t writeIntegrationPointData(std::vector<char>& data) = 0;
+    virtual void readIntegrationPointData(std::vector<char> const& data) = 0;
```

Comment:
Same as with extrapolators; If one does not derive from `IntegrationPointSerialization` then nothing happens, otherwise there is a commitment to write and read the IPData.